src/lib.rs

```diff
@@ -10,47 +10,24 @@ use core::cmp::Ordering;
 use lset::{Empty, Line, Span};
 use primordial::{Address, Offset, Page};
 
-bitflags::bitflags! {
-    /// Memory access permissions.
-    #[derive(Default)]
-    #[repr(transparent)]
-    pub struct Access: usize {
-        /// Read access
-        const READ = 1 << 0;
-
-        /// Write access
-        const WRITE = 1 << 0;
-
-        /// Execute access
-        const EXECUTE = 1 << 0;
-    }
-}
-
-/// A ledger record.
+/// A ledger region.
 #[derive(Copy, Clone, Debug, Default, PartialEq, Eq)]
-pub struct Record {
-    /// The covered region of memory.
-    pub region: Line<Address<usize, Page>>,
-
-    /// The access permissions.
-    pub access: Access,
-
-    length: usize,
+pub struct Region<T> {
+    /// Limits for the region.
+    pub limits: Line<Address<usize, Page>>,
+    /// Fill value for the region.
+    pub value: Option<T>,
```

Comment:
This isn't a great data type. It cannot be naturally aligned.

Reply:
I agree with the aligned part, i.e. it needs to be concrete type. I got that. What is a glitch in earlier type is that it is storing ledger's fields in every region. Length field makes sense only for the first entry.

Reply:
I.e. padding area is meant for padding.

Reply:
Now token is 32 bytes, and all constructors, include two `Region` are `const` constructors, which makes creating objects much more enjoyable experience (before it was only `Ledger`). A token is arbitrary `u64`.